app/src/main/java/de/danoeh/antennapod/dialog/ShareDialog.java

```diff
@@ -0,0 +1,121 @@
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.util.ShareUtils;
+
+public class ShareDialog extends DialogFragment {
+
+    private static final String ARGUMENT_FEED_ITEM = "feedItem";
+
+    private static final String TAG = "ShareDialog";
+    private Context ctx;
+    private FeedItem item;
+
+    private static final String PREF_SHARE_DIALOG_OPTION = "prefShareDialogOption";
+    private static final String PREF_SHARE_EPISODE_START_AT = "prefShareEpisodeStartAt";
+
+    private RadioGroup radioGroup;
+    private RadioButton radioEpisodeWebsite;
+    private RadioButton radioMediaFile;
+    private CheckBox checkBoxStartAt;
+    private SharedPreferences prefs;
+
+    public ShareDialog() {
+        // Empty constructor required for DialogFragment
+    }
+
+    public static ShareDialog newInstance(FeedItem item) {
+        Bundle arguments = new Bundle();
+        arguments.putSerializable(ARGUMENT_FEED_ITEM, item);
+        ShareDialog dialog = new ShareDialog();
+        dialog.setArguments(arguments);
+        return dialog;
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
+        if (getArguments() != null) {
+            ctx = getActivity();
+            item = (FeedItem) getArguments().getSerializable(ARGUMENT_FEED_ITEM);
+            prefs = getActivity().getSharedPreferences("ShareDialog", Context.MODE_PRIVATE);
+        }
+
+        View content = View.inflate(ctx, R.layout.share_episode_dialog, null);
+        AlertDialog.Builder builder = new AlertDialog.Builder(ctx);
+        builder.setTitle(R.string.share_label);
+        builder.setView(content);
+
+        radioGroup = content.findViewById(R.id.share_dialog_radio_group);
+        radioEpisodeWebsite = content.findViewById(R.id.share_episode_website_radio);
+        radioMediaFile = content.findViewById(R.id.share_media_file_radio);
+        checkBoxStartAt = content.findViewById(R.id.share_start_at_timer_dialog);
+
+        setupOptions();
+
+        builder.setPositiveButton(R.string.share_label, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int id) {
+                boolean includePlaybackPosition = checkBoxStartAt.isChecked();
+                if (radioEpisodeWebsite.isChecked()) {
+                    ShareUtils.shareFeedItemLink(ctx, item, includePlaybackPosition);
+                    prefs.edit().putString(PREF_SHARE_DIALOG_OPTION, "website").apply();
+                } else {
+                    ShareUtils.shareFeedItemDownloadLink(ctx, item, includePlaybackPosition);
+                    prefs.edit().putString(PREF_SHARE_DIALOG_OPTION, "media").apply();
+                }
+                prefs.edit().putBoolean(PREF_SHARE_EPISODE_START_AT, includePlaybackPosition).apply();
+            }
+        })
+        .setNegativeButton(R.string.cancel_label, new DialogInterface.OnClickListener() {
```

Comment:
Checkstyle complains here. If you just remove the line break between `)` and `.`, it is happy :)

Reply:
fixed :)